Fix fail test in `utils`

### Which package has an issue?

@yamada-ui/react

### Package Version

1.3.3

### Description

In what follows, an error occurs as a result of test.


```bash
AssertionError: expected undefined to be 'success' // Object.is equality

Ignored nodes: comments, script, style
<html>
  <head />
  <body>
    <div />
  </body>
</html>

- Expected: 
"success"

+ Received: 
undefined

 ❯ packages/utils/tests/react.test.tsx:286:38
    284|       })
    285|       await waitFor(async () => {
    286|         expect(result.current.value).toBe("success")
       |                                      ^
    287|       })
    288|     })

```

### Link to Reproduction

_No response_

### Steps to Reproduce

1. run 'pnpm test'
2. See error


### Browser

None

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

Yes

### Additional Information

None.

--- a/packages/components/modal/tests/dialog.test.tsx
+++ b/packages/components/modal/tests/dialog.test.tsx
@@ -1,5 +1,5 @@
 import { Button, useDisclosure } from "@yamada-ui/react"
-import { a11y, render, waitFor, screen } from "@yamada-ui/test"
+import { a11y, render, waitFor } from "@yamada-ui/test"
 import { Dialog, DialogHeader, DialogFooter, DialogBody } from "../src"
 
 describe("<Dialog />", () => {
@@ -91,7 +91,9 @@ describe("<Dialog />", () => {
   })
 
   test("Dialog renders correctly when close", async () => {
-    const { user, findByTestId, getByText } = render(<DialogCloseExample />)
+    const { user, findByTestId, queryByTestId, getByText } = render(
+      <DialogCloseExample />,
+    )
 
     const openDialogButton = await findByTestId("OpenDialog")
 
@@ -99,23 +101,23 @@ describe("<Dialog />", () => {
     await expect(findByTestId("Dialog")).resolves.toBeInTheDocument()
     const cancelDialogButton = getByText("dialog-cancel")
     await user.click(cancelDialogButton)
-    waitFor(() => {
-      expect(screen.findByTestId("Dialog")).toBeNull()
+    await waitFor(async () => {
+      expect(queryByTestId("Dialog")).toBeNull()
     })
     await user.click(openDialogButton)
     await expect(findByTestId("Dialog")).resolves.toBeInTheDocument()
     const otherDialogButton = getByText("dialog-other")
     await user.click(otherDialogButton)
-    waitFor(() => {
-      expect(screen.findByTestId("Dialog")).toBeNull()
+    await waitFor(async () => {
+      expect(queryByTestId("Dialog")).toBeNull()
     })
 
     await user.click(openDialogButton)
     await expect(findByTestId("Dialog")).resolves.toBeInTheDocument()
     const successDialogButton = getByText("dialog-success")
     await user.click(successDialogButton)
-    waitFor(() => {
-      expect(screen.findByTestId("Dialog")).toBeNull()
+    await waitFor(async () => {
+      expect(queryByTestId("Dialog")).toBeNull()
     })
   })
 })

--- a/packages/utils/tests/react.test.tsx
+++ b/packages/utils/tests/react.test.tsx
@@ -278,12 +278,12 @@ describe("React", () => {
       }
       const { result } = renderHook(() => useAsyncRetry(asyncFunction, []))
       expect(result.current.loading).toBeTruthy()
-      await waitFor(() => {})
-      act(() => {
+
+      act(async () => {
         result.current.retry()
-      })
-      await waitFor(() => {
-        expect(result.current.value).toBe("success")
+        await waitFor(() => {
+          expect(result.current.value).toBe("success")
+        })
       })
     })
   })

